tools/azsdk-cli/docs/specs/6-package-updater.spec.md

```diff
@@ -0,0 +1,697 @@
+
+Outputs:
+
+```json
+{"result":"succeeded", "changed_files":["..\\pom.xml","C:\\dev\\repos\\azure-sdk-for-java\\pom.xml"], "changed_count": 2, "message":"All relevant pom.xml files have been updated."}
+```
+
+```json
+{"result":"failed","message":"API export failed: missing dotnet tool", "next_steps": "call verify-setup tool"}
+```
+
+Failure Modes:
+
+- Missing expected file (e.g., `_metadata.json`) → report error.
+
+##### Update Metadata script
+
+When the tool executes the language-specific `update-metadata` script it will pass the following parameters. All paths are absolute.
+
+| Parameter | Description | Type | Required |
+|-----------|-------------|-----:|:--------:|
+| `SdkRepoPath` | Absolute path to the root folder of the local SDK repository. | string | Yes |
+| `PackagePath` | Absolute path to the root folder of the local SDK project (package). | string | Yes |
+
+**Example invocations:**
+
+```powershell
+update-metadata.ps1 \
+      -SdkRepoPath C:\dev\repos\azure-sdk-for-net \
+      -PackagePath C:\dev\repos\azure-sdk-for-net\services\newservice
+```
+
+#### 4. CI Update Tool
+
+Name (CLI): `azsdk package update-ci`
+
+Name (MCP): `azsdk_package_update_ci`
+
+> **Note:** Support for this tool is planned for post‑milestone-1.
+
+Purpose: Create or update CI pipeline configuration.
+
+Inputs:
+
+- `--package-path <abs path>` (required) – Absolute path to the root of the SDK package.
+
+Execution Steps:
+
+1. Create the corresponding YAML file, like `ci.yml`, which will be used to setup pipeline.
+2. Update the relevant CI YAML file accordingly
+3. Return the changed file name with the result.
+
+Outputs (examples):
+
+```json
+{"result":"succeeded", "ci-file": "../ci.yml", "message":"CI files created: ci.yml", "next_steps": "update changelog"}
+```
+
+```json
+{"result":"failed", "message":"script exited 1: missing pwsh"}
+```
+
+Failure Modes:
+
+- Create file fails due to insufficient permissions → return `failed` with error.
+
+##### Update CI script
+
+When the tool executes the language-specific `update-ci` script it will pass the following parameters. All paths are absolute.
+
+| Parameter | Description | Type | Required |
+|-----------|-------------|-----:|:--------:|
+| `SdkRepoPath` | Absolute path to the root folder of the local SDK repository. | string | Yes |
+| `PackagePath` | Absolute path to the root folder of the local SDK project (package). | string | Yes |
+
+**Example invocations:**
+
+```powershell
+update-ci.ps1 \
+      -SdkRepoPath C:\dev\repos\azure-sdk-for-net \
+      -PackagePath C:\dev\repos\azure-sdk-for-net\services\newservice
+```
+
+#### 5. Aggregated Tool (chained execution)
+
+Name (CLI): `azsdk package update-all`
```

Comment:
May I know the reason for having both separate tool and aggreated tool?

Reply:
Aggregated tool is expected to integrate to automation pipeline scenario and release scenario.